src/Ng/Jobs/MonitoringProcessorJob.cs

```diff
@@ -0,0 +1,201 @@
+            {
+                throw new ArgumentException("File storage is not supported!");
+            }
+
+            var monitoringStorageFactory = CommandHelpers.CreateStorageFactory(arguments, verbose);
+            var auditingStorageFactory = CommandHelpers.CreateSuffixedStorageFactory("Auditing", arguments, verbose);
+
+            var endpointInputs = CommandHelpers.GetEndpointFactoryInputs(arguments);
+
+            var messageHandlerFactory = CommandHelpers.GetHttpMessageHandlerFactory(verbose);
+
+            Logger.LogInformation(
+                "CONFIG gallery: {Gallery} index: {Index} storage: {Storage} auditingStorage: {AuditingStorage} endpoints: {Endpoints}",
+                gallery, index, monitoringStorageFactory, auditingStorageFactory, string.Join(", ", endpointInputs.Select(e => e.Name)));
+
+            _packageValidator = new PackageValidatorFactory(LoggerFactory)
+                .Create(gallery, index, auditingStorageFactory, endpointInputs, messageHandlerFactory, verbose);
+
+            _queue = CommandHelpers.CreateStorageQueue<PackageValidatorContext>(arguments);
+
+            _statusService = CommandHelpers.GetPackageMonitoringStatusService(arguments, monitoringStorageFactory, LoggerFactory);
+
+            _notificationService = new LoggerMonitoringNotificationService(LoggerFactory.CreateLogger<LoggerMonitoringNotificationService>());
+
+            _regResource = Repository.Factory.GetCoreV3(index).GetResource<RegistrationResourceV3>(cancellationToken);
+
+            _client = new CollectorHttpClient(messageHandlerFactory());
+        }
+
+        protected override async Task RunInternal(CancellationToken cancellationToken)
+        {
+            await ParallelAsync.Repeat(() => ProcessPackages(cancellationToken));
+        }
+
+        private async Task ProcessPackages(CancellationToken token)
+        {
+            IStorageQueueMessage<PackageValidatorContext> queueMessage = null;
+            do
+            {
+                Logger.LogInformation("Fetching next queue message.");
+                queueMessage = await _queue.GetNextAsync(token);
+                await HandleQueueMessage(queueMessage, token);
+            } while (queueMessage != null);
+
+            Logger.LogInformation("No messages left in queue.");
+        }
+
+        private async Task HandleQueueMessage(IStorageQueueMessage<PackageValidatorContext> queueMessage, CancellationToken token)
+        {
+            if (queueMessage == null)
+            {
+                return;
+            }
+
+            var queuedContext = queueMessage.Contents;
+
+            try
+            {
+                await RunPackageValidator(queuedContext, token);
+
+                await _queue.Remove(queueMessage, token);
+            }
+            catch (Exception e)
+            {
+                await SaveFailedPackageMonitoringStatus(queuedContext, e, token);
+
+                await _queue.Remove(queueMessage, token);
+            }
```

Comment:
move `await _queue.Remove(queueMessage, token);` outside `try..catch`

Reply:
got it